flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/ObjectArrayComparator.java

```diff
@@ -0,0 +1,199 @@
+
+	private static final long serialVersionUID = 1L;
+
+	private transient T[] reference;
+
+	protected final boolean ascendingComparison;
+
+	private final TypeSerializer<T[]> serializer;
+
+	private TypeComparator<? super Object> comparatorInfo;
+
+	// For use by getComparators
+	@SuppressWarnings("rawtypes")
+	private final TypeComparator[] comparators = new TypeComparator[] {this};
+
+	public ObjectArrayComparator(boolean ascending, TypeSerializer<T[]> serializer, TypeComparator<? super Object> comparatorInfo) {
+		this.ascendingComparison = ascending;
+		this.serializer = serializer;
+		this.comparatorInfo = comparatorInfo;
+	}
+
+	@Override
+	public void setReference(T[] reference) {
+		this.reference = reference;
+	}
+
+	@Override
+	public boolean equalToReference(T[] candidate) {
+		return compare(this.reference, candidate) == 0;
+	}
+
+	@Override
+	public int compareToReference(TypeComparator<T[]> referencedComparator) {
+		int comp = compare(((ObjectArrayComparator<T,C>) referencedComparator).reference, reference);
+		return comp;
+	}
+
+	@Override
+	public int compareSerialized(DataInputView firstSource, DataInputView secondSource) throws IOException {
+		T[] firstArray = serializer.deserialize(firstSource);
+		T[] secondArray = serializer.deserialize(secondSource);
+
+		int comp = compare(firstArray, secondArray);
+		return comp;
+	}
+
+	@Override
+	public int extractKeys(Object record, Object[] target, int index) {
+		target[index] = record;
+		return 1;
+	}
+
+	@Override
+	public TypeComparator[] getFlatComparators() {
+		return comparators;
+	}
+
+	@Override
+	public boolean supportsNormalizedKey() {
+		return false;
+	}
+
+	@Override
+	public boolean supportsSerializationWithKeyNormalization() {
+		return false;
+	}
+
+	@Override
+	public int getNormalizeKeyLen() {
+		return 0;
+	}
+
+	@Override
+	public boolean isNormalizedKeyPrefixOnly(int keyBytes) {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public void putNormalizedKey(T[] record, MemorySegment target, int offset, int numBytes) {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public void writeWithKeyNormalization(T[] record, DataOutputView target) throws IOException {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public T[] readWithKeyDenormalization(T[] reuse, DataInputView source) throws IOException {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public boolean invertNormalizedKey() {
+		return !ascendingComparison;
+	}
+
+	@Override
+	public int hash(T[] record) {
+		return Arrays.hashCode(record);
+	}
+
+	private int compareValues(Object first, Object second) {
+		/**
+		 * uses the chosen comparator ( of primitive or composite type ) & compares the provided objects as input
+		 */
+		return comparatorInfo.compare(first, second);
+	}
+
+	@SuppressWarnings("unchecked")
+	private int parseGenericArray(Object firstArray, Object secondArray) {
```

Comment:
I think this is not the right way to go. Simply check then length of the arrays in the `compare` method and then call for all the elements the type comparator of the element type of the array.